app/src/main/java/com/stevesoltys/seedvault/restore/install/ApkRestore.kt

```diff
@@ -37,15 +38,22 @@ internal class ApkRestore(
 ) {
 
     private val pm = context.packageManager
+    private val restrictions = context.getSystemService(UserManager::class.java).userRestrictions
 
     @Suppress("BlockingMethodInNonBlockingContext")
     fun restore(backup: RestorableBackup) = flow {
+        val cantInstallApks = restrictions.getBoolean(UserManager.DISALLOW_INSTALL_APPS,
+            false) ||
+            restrictions.getBoolean(UserManager.DISALLOW_INSTALL_UNKNOWN_SOURCES,
+            false) ||
+            restrictions.getBoolean(UserManager.DISALLOW_INSTALL_UNKNOWN_SOURCES_GLOBALLY,
+            false)
         // filter out packages without APK and get total
         val packages = backup.packageMetadataMap.filter {
             // We also need to exclude the DocumentsProvider used to retrieve backup data.
             // Otherwise, it gets killed when we install it, terminating our restoration.
             val isStorageProvider = it.key == storagePlugin.providerPackageName
-            it.value.hasApk() && !isStorageProvider
+            it.value.hasApk() && !isStorageProvider && !cantInstallApks
```

Comment:
I'm sure there's a more elegant way to do this.

Maybe check this, and then init an empty `MutableInstallResult`, set to finish and return that?

Reply:
To do what? If we can not install any APK, we don't need to do any filtering, right?

We could probably emit an install result right away with all apps as failed (if they are not already installed)?

Reply:
> To do what? If we can not install any APK, we don't need to do any filtering, right?

Yes

> We could probably emit an install result right away with all apps as failed (if they are not already installed)?

Yes that was the other option I considered.